s/sentencepiece/sentencepiece_0.2.0_ubi_9.3.sh

```diff
@@ -0,0 +1,174 @@
+#                    package and/or distribution. In such case, please
+#                    contact "Maintainer" of this script.
+#
+# ---------------------------------------------------------------------------
+
+PACKAGE_NAME=sentencepiece
+PACKAGE_VERSION=${1:-v0.2.0}
+PACKAGE_URL=https://github.com/google/sentencepiece.git
+PACKAGE_DIR=sentencepiece/python
+
+yum install -y make libtool cmake git wget xz zlib-devel openssl-devel bzip2-devel libffi-devel libevent-devel patch python python-devel ninja-build gcc-toolset-13  pkg-config
+
+dnf install -y gcc-toolset-13-libatomic-devel
+
+export PATH=/opt/rh/gcc-toolset-13/root/usr/bin:$PATH
+export LD_LIBRARY_PATH=/opt/rh/gcc-toolset-13/root/usr/lib64:$LD_LIBRARY_PATH
+export SITE_PACKAGE_PATH="/lib/python3.12/site-packages"
+
+SCRIPT_DIR=$(pwd)
+
+#building abesil-cpp and libprotobuf 
+
+pip install --upgrade cmake pip setuptools wheel ninja packaging pytest
+
+#Building abseil-cpp
+ABSEIL_VERSION=20240116.2
+ABSEIL_URL="https://github.com/abseil/abseil-cpp"
+mkdir $SCRIPT_DIR/abseil-prefix
+PREFIX=$SCRIPT_DIR/abseil-prefix
+
+git clone $ABSEIL_URL -b $ABSEIL_VERSION
+echo "abseil-cpp build starts"
+cd abseil-cpp
+
+SOURCE_DIR=$(pwd)
+
+mkdir -p $SOURCE_DIR/local/abseilcpp
+abseilcpp=$SOURCE_DIR/local/abseilcpp
+
+mkdir build
+cd build
+
+cmake -G Ninja \
+    ${CMAKE_ARGS} \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_CXX_STANDARD=17 \
+    -DCMAKE_INSTALL_LIBDIR=lib \
+    -DCMAKE_INSTALL_PREFIX=${PREFIX} \
+    -DBUILD_SHARED_LIBS=ON \
+    -DABSL_PROPAGATE_CXX_STD=ON \
+    -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
+   ..
+cmake --build .
+cmake --install .
+
+cd $SCRIPT_DIR
+cp -r  $PREFIX/* $abseilcpp/
+
+echo "------------abseil-cpp installed--------------"
+
+#Setting paths and versions
+#PREFIX=$SITE_PACKAGE_PATH
+ABSEIL_PREFIX=$SOURCE_DIR/local/abseilcpp
+
+export C_COMPILER=$(which gcc)
+export CXX_COMPILER=$(which g++)
+
+#Build libprotobuf
+git clone https://github.com/protocolbuffers/protobuf
+cd protobuf
+git checkout v4.25.3
+
+LIBPROTO_DIR=$(pwd)
+mkdir -p $LIBPROTO_DIR/local/libprotobuf
+LIBPROTO_INSTALL=$LIBPROTO_DIR/local/libprotobuf
+
+git submodule update --init --recursive
+rm -rf ./third_party/googletest | true
+rm -rf ./third_party/abseil-cpp | true
+
+cp -r $SCRIPT_DIR/abseil-cpp ./third_party/
+
+mkdir build
+cd build
+
+cmake -G "Ninja" \
+   ${CMAKE_ARGS} \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_CXX_STANDARD=17 \
+    -DCMAKE_C_COMPILER=$C_COMPILER \
+    -DCMAKE_CXX_COMPILER=$CXX_COMPILER \
+    -DCMAKE_INSTALL_PREFIX=$LIBPROTO_INSTALL \
+    -Dprotobuf_BUILD_TESTS=OFF \
+    -Dprotobuf_BUILD_LIBUPB=OFF \
+    -Dprotobuf_BUILD_SHARED_LIBS=ON \
+    -Dprotobuf_ABSL_PROVIDER="module" \
+    -DCMAKE_PREFIX_PATH=$ABSEIL_PREFIX \
+    -Dprotobuf_JSONCPP_PROVIDER="package" \
+    -Dprotobuf_USE_EXTERNAL_GTEST=OFF \
+    ..
+
+cmake --build . --verbose
+cmake --install .
+
+cd $SCRIPT_DIR
+ 
+echo "------------ libprotobuf installed--------------"
+
```

Comment:
In opence script we are also using protobuf at build time. Could you please add that too.